### DISCUSSION TITLE
Upgrade of the Community RPC Experimental Program and Expansion to the Community Infrastructure Program

### DISCUSSION THREAD

**Discussion by hackfisher** [ringdao-community-guild/discussion#132] [Infrastructure]
The [Community RPC Experimental Program](https://snapshot.org/#/darwiniacommunitydao.eth/proposal/0x26be609917096095e8fb2a617d6def3fa62b3240285a9887bcd9184afb2a0ddd) proposed by @fewensa has been operational for over 3 quarters, yet the participation of community nodes remains insufficient. Therefore, at the recent Steering Committee Meeting, participants suggested modifications and upgrades to the plan. The main proposals include:

- Rename the program to the "Community Infrastructure Program" to broaden its scope beyond just RPC setups, including support for explorers based on open-source solutions or constructing higher-performance RPCs based on new approaches.
- Adjust the incentives for community RPC nodes to a maximum of 50 USD per node per month.
- Keep the budget unchanged and use the surplus funds to support and develop other infrastructure.
- Better support and complement the current integration strategy, which focuses on the Darwinia. By using Crab, aim to utilize only free integrations and facilities to save funds for more paid facilities for Darwinia.
- Create a new "Infrastructure" category on the forum to manage and track the progress and updates of this program. The program will continue to be led by @fewensa , a member of DCDAO and a senior DevSecOps engineer from Itering.
- Outline some preliminary directions for funding development:
  - Higher-performance query RPCs, with candidate open solutions including: 
    - Adjusting load balancing to IP Hash mode to address instability issues in indexers; 
    - Researching and testing [FlumeLight](https://blog.rivet.cloud/polygon/flumelight/)
    - Referring to [Local HyperRPC](https://github.com/enviodev/local-hyperrpc) for architectural optimization of RPC services.
    - New indexing solutions like Ponder and [envio](https://envio.dev/)

  - Deploying Explorer based on open-source solutions, such as Blockscout. 
    - Given that transactions on the network are now primarily EVM-based, it is considered to build an Explorer based on the Blockscout open-source solution, while advancing the integration of [Sourcify](https://github.com/orgs/dcdao/discussions/116). These explorers can be used as backup or default for current explorers, which users can find the EVM txs and token more easily.
    - Considering the recent [proposal](https://darwinia2.subsquare.io/referenda/11) from Subscan, the costs for Explorers for Crab and Pangolin are not low and conflict with the aforementioned integration strategy. 
    - For Extrinsics data needed for developer debugging, switch to using tools like [Subsquid](https://docs.subsquid.io/subsquid-network/reference/substrate-networks/) and [Appsmith](https://app.appsmith.com/).
    
    
Reference:
1. https://github.com/0xPolygon/chain-indexer-framework
2. https://routescan.io/
3. https://rpcs-test.vercel.app/
4. https://crab.subview.xyz/
5. https://github.com/fuzzland/cached-eth-rpc

**Comment by fewensa** [comment]
I think this is good. I will launch a new proposal to describe the upgrade of the service. I think this upgrade focuses on "infrastructure", so the actual maintenance projects may not be limited to some of the work described above, and there may be more in the future. There are other possibilities.
